### ISSUE TITLE
ValueError: The name "sequential_1" is used 2 times in the model. All layer names should be unique.

### ISSUE THREAD

**Issue by uttamdhakal** [keras-team/keras#12187]
I am trying to combine my saved model to the current model and I get:

> ValueError: The name "sequential_1" is used 2 times in the model. All layer names should be unique.

I have tried to rename the layer with following piece of code but without any success
```
for layer in model.layers:
    layer.name = layer.name + str("_2")
```

Both the current model and saved model are actually identical, I have however, trained the saved model on the different dataset. This is my code:

```
saved_model = load_model('Models/saved_model_books.h5')

for layer in saved_model.layers:
    layer.trainable = False

model = Sequential()
model.add(Embedding(max_words, embedding_dim, input_length=maxlen))
model.add(Bidirectional(LSTM(32)))
model.add(Dropout(0.6))
model.add(Dense(2, activation='sigmoid'))

model.layers[0].set_weights([embedding_matrix])
model.layers[0].trainable = False

for layer in model.layers:
    layer.name = layer.name + str("_2")

commonInput = Input((100,))

out1 = saved_model(commonInput)
out2 = model(commonInput)

merged = Add()([out1,out2])
newModel = Model(commonInput, merged)

newModel.compile(optimizer='rmsprop', loss='binary_crossentropy',metrics=['acc'])

history = newModel.fit(sequences, labels, epochs=10, batch_size=32, validation_split=0.2)
```

The error is caused by the line: 

> newModel = Model(commonInput, merged)

How can I resolve the error? Or is there any better way to do this?

**Comment by emc5ud** [comment]
This worked for me. I simply changed the model name rather than each layer name. 
```
m1 = keras.models.load_model('../resources/models/baseline/baseline_0.h5')
m1.name = 'test1'
m2 = keras.models.load_model('../resources/models/baseline/baseline_0.h5')
m2.name = 'test2'
commonInput = Input((128, 128, 9))
out1 = m1(commonInput)
out2 = m2(commonInput)
merged = keras.layers.Add()([out1,out2])
newModel = Model(commonInput, merged)
```

**Comment by uttamdhakal** [comment]
Thanks, this works. One more question, do you happen to know the difference between add and concatenate? When would one use what? I have seen the definitions but I would love the answer in plain english.

**Comment by emc5ud** [comment]
I know you said you've seen the definitions, but just I'll give you this link  just in case https://keras.io/layers/merge/

Say you have two tensors `t11 and `t2` shape [? , 20, 20, 3]

`keras.layers.Add()([t1, t2]` yields a tensor shape [?, 20, 20, 3]

'keras.layers.Concatenate(axis=-1)[t1, t2]` yeilds a tensor shaped [?, 20, 20, 6] 

Add, Subtract, Multiply, etc always expects the inputs and outputs to have the same shape

Concatenate yeilds a tensor where the new shape depends on the axis you chose to concatenate over. 

Hope this helps

**Comment by xuzhang5788** [comment]
> This worked for me. I simply changed the model name rather than each layer name.
> 
> ```
> m1 = keras.models.load_model('../resources/models/baseline/baseline_0.h5')
> m1.name = 'test1'
> m2 = keras.models.load_model('../resources/models/baseline/baseline_0.h5')
> m2.name = 'test2'
> commonInput = Input((128, 128, 9))
> out1 = m1(commonInput)
> out2 = m2(commonInput)
> merged = keras.layers.Add()([out1,out2])
> newModel = Model(commonInput, merged)
> ```
> 
> without changing the names I got a similar error

**Comment by xuzhang5788** [comment]
@emc5ud Thank you so much. I took several hours to solve this problem. Fortunately, I found your post and it worked. Thank you so so much!

**Comment by WrathofBhuvan11** [comment]
> This worked for me. I simply changed the model name rather than each layer name.
> 
> ```
> m1 = keras.models.load_model('../resources/models/baseline/baseline_0.h5')
> m1.name = 'test1'
> m2 = keras.models.load_model('../resources/models/baseline/baseline_0.h5')
> m2.name = 'test2'
> commonInput = Input((128, 128, 9))
> out1 = m1(commonInput)
> out2 = m2(commonInput)
> merged = keras.layers.Add()([out1,out2])
> newModel = Model(commonInput, merged)
> ```
> 
> without changing the names I got a similar error

I'm getting this error
AttributeError: Can't set the attribute "name", likely because it conflicts with an existing read-only @property of the object. Please choose a different name.
